OSDOCS#10130: Doc support for ARM control plane with x86 compute nodes

sr1kar99 opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s):
4.16
<!--- Specify the version or versions of OpenShift your PR applies to. -->

Issue:
[OSDOCS-10130](https://issues.redhat.com/browse/OSDOCS-10130)
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview:
- [Creating a cluster with multi-architecture compute machine on Azure](https://76246--ocpdocs-pr.netlify.app/openshift-enterprise/latest/post_installation_configuration/configuring-multi-arch-compute-machines/creating-multi-arch-compute-nodes-azure)
- [Creating a cluster with multi-architecture compute machines on AWS](https://76246--ocpdocs-pr.netlify.app/openshift-enterprise/latest/post_installation_configuration/configuring-multi-arch-compute-machines/creating-multi-arch-compute-nodes-aws)
- [Creating a cluster with multi-architecture compute machines on GCP](https://76246--ocpdocs-pr.netlify.app/openshift-enterprise/latest/post_installation_configuration/configuring-multi-arch-compute-machines/creating-multi-arch-compute-nodes-gcp)
- [Creating a cluster with multi-architecture compute machines on bare metal, IBM Power, or IBM Z](https://76246--ocpdocs-pr.netlify.app/openshift-enterprise/latest/post_installation_configuration/configuring-multi-arch-compute-machines/creating-multi-arch-compute-nodes-bare-metal)

<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

sr1kar99 commented:
Hi @jeffdyoung 
Could you please review this PR?
Thanks!

jeffdyoung commented on modules/multi-architecture-modify-machine-set-aws.adoc:
```diff
@@ -4,26 +4,23 @@
-= Adding an ARM64 compute machine set to your cluster
-
-To configure a cluster with multi-architecture compute machines, you must create a AWS ARM64 compute machine set. This adds ARM64 compute nodes to your cluster so that your cluster has multi-architecture compute machines.
+To configure an AWS cluster with multi-architecture compute machines, you must create an AWS ARM64 or AMD64 compute machine set. This adds ARM64 or AMD64 compute nodes to your cluster.
```
Instead of **AMD64** and **ARM64**, we prefer to use **64-Bit ARM** and **64-bit x86**

jeffdyoung replied:
also, you can remove: `This adds ARM64 or AMD64 compute nodes to your cluster`

sr1kar99 replied:
Updated as per the suggestion.

jeffdyoung commented on modules/multi-architecture-modify-machine-set.adoc:
```diff
@@ -4,23 +4,24 @@
+= Adding a multi-architecture compute machine set to your Azure cluster
 
-= Adding a multi-architecture compute machine set to your cluster
+To configure an Azure cluster with multi-architecture compute machines, you must create an Azure ARM64 or AMD64 compute machine set that uses the ARM64 or AMD64 boot image. This adds ARM64 or AMD64 compute nodes to your cluster.
```
remove `This adds ARM64 or AMD64 compute nodes to your cluster.`

sr1kar99 replied:
Updated as per the suggestion.

jeffdyoung commented on modules/multi-architecture-modify-machine-set-gcp.adoc:
```diff
@@ -4,25 +4,23 @@
+= Adding a multi-architecture compute machine set to your GCP cluster
 
-To configure a cluster with multi-architecture compute machines, you must create a GCP ARM64 compute machine set. This adds ARM64 compute nodes to your cluster.
+To configure a GCP cluster with multi-architecture compute machines, you must create a GCP ARM64 or AMD64 compute machine set. This adds ARM64 or AMD64 compute nodes to your cluster.
```
remove `This adds ARM64 or AMD64 compute nodes to your cluster.`

sr1kar99 commented:
> @sr1kar99 only a few small things, otherwise LGTM. The only times we use `arm64/aarch64/amd64/x86_64` is when a command's syntax requires it, or if a cloud provider uses a specific term. Otherwise use `64-bit ARM` and `64-bit x86`.

Hi @jeffdyoung 
Thanks for the review. Updated as per the suggestions.

In the "Adding a multi-architecture compute machine set to your Azure cluster" module, we have the following line:

"To configure an Azure cluster with multi-architecture compute machines, you must create an Azure 64-bit ARM or 64-bit x86 compute machine set that uses the 64-bit ARM or 64-bit x86 boot image."

Though we have added support for `64-bit x86` arch, we don't have a procedure on "Creating a 64-bit x86 boot image using the Azure image gallery" similar to the existing "Creating a 64-bit ARM boot image using the Azure image gallery" proc.

Should we modify the existing "Creating a 64-bit ARM boot image using the Azure image gallery" proc to include the support for `64-bit x86` arch or add a separate module?

sr1kar99 commented:
@aleskandro 
Could you please review this PR?
Thanks!

aleskandro commented on modules/multi-architecture-modify-machine-set-aws.adoc:
```diff
@@ -4,26 +4,23 @@
-= Adding an ARM64 compute machine set to your cluster
-
-To configure a cluster with multi-architecture compute machines, you must create a AWS ARM64 compute machine set. This adds ARM64 compute nodes to your cluster so that your cluster has multi-architecture compute machines.
+To configure an AWS cluster with multi-architecture compute machines, you must create an AWS 64-bit ARM or 64-bit x86 compute machine set.
```
This incipit does not give a so meaningful information now that the secondary architecture can be any. Perhaps, something like 
> When your cluster relies on the multi-architecture payload, you can add nodes with diverse architectures. For example, you can add amd64 workers alongside other arm64 workers in a cluster using arm64 control plane machines. In this case, we define the secondary architecture as `amd64`.

sr1kar99 replied:
Updated as follows:

"After creating a multi-architecture cluster, you can add nodes with different architectures. For example, you can add 64-bit ARM compute machines to a cluster that uses 64-bit ARM control plane machines and already includes 64-bit x86 compute machines. In this case, 64-bit ARM is considered the secondary architecture."

aleskandro commented on post_installation_configuration/configuring-multi-arch-compute-machines/creating-multi-arch-compute-nodes-aws.adoc:
```diff
@@ -6,13 +6,14 @@ include::_attributes/common-attributes.adoc[]
 toc::[]
 
-To create an AWS cluster with multi-architecture compute machines, you must first create a single-architecture AWS installer-provisioned cluster with the multi-architecture installer binary. For more information on AWS installations, refer to xref:../../installing/installing_aws/ipi/installing-aws-customizations.adoc#installing-aws-customizations[Installing a cluster on AWS with customizations].  ...
+To create an AWS cluster with multi-architecture compute machines, you must first create a single-architecture AWS installer-provisioned cluster with the multi-architecture installer binary. For more information on AWS installations, see xref:../../installing/installing_aws/ipi/installing-aws-customizations.adoc#installing-aws-customizations[Installing a cluster on AWS with customizations]. You c ...
```
```suggestion
To create an AWS cluster with multi-architecture compute machines, you must first create a single-architecture AWS installer-provisioned cluster with the multi-architecture installer binary or migrate your existing cluster to the multi-architecture payload. For more information on AWS installations, see xref:../../installing/installing_aws/ipi/installing-aws-customizations.adoc#installing-aws-customizations[Installing a cluster on AWS with customizations]. You can then add a 64-bit ARM or 64-bit x86 compute machine set to your AWS cluster.
```

[Slightly OT] Should we clarify this?

sr1kar99 replied:
Makes sense.
Updated as per the suggestion.

aleskandro commented on post_installation_configuration/configuring-multi-arch-compute-machines/creating-multi-arch-compute-nodes-azure.adoc:
```diff
@@ -6,9 +6,9 @@ include::_attributes/common-attributes.adoc[]
+To deploy an Azure cluster with multi-architecture compute machines, you must first create a single-architecture Azure installer-provisioned cluster that uses the multi-architecture installer binary. For more information on Azure installations, see xref:../../installing/installing_azure/installing-azure-customizations.adoc#installing-azure-customizations[Installing a cluster on Azure with customi ...
 
-The following procedures explain how to generate an ARM64 boot image and create an Azure compute machine set that uses the ARM64 boot image. This adds ARM64 compute nodes to your cluster and deploys the amount of ARM64 virtual machines (VM) that you need.
+For a cluster with 64-bit ARM compute machines, the following procedures explain how to generate a 64-bit ARM boot image and create an Azure compute machine set that uses the 64-bit ARM boot image. You can add compute nodes to your cluster and deploy the 64-bit ARM or 64-bit x86 virtual machines (VM).
```
```suggestion
For a cluster with 64-bit AMD compute machines, the following procedures explain how to generate a 64-bit ARM boot image and create an Azure compute machine set that uses the 64-bit ARM boot image. You can add compute nodes to your cluster and deploy the 64-bit ARM or 64-bit x86 virtual machines (VM).
```

sr1kar99 replied:
Updated as per the suggestion.

sr1kar99 commented on modules/multi-architecture-modify-machine-set-aws.adoc:
```diff
@@ -89,15 +86,15 @@ spec:
 <2> Specify the infrastructure ID, role node label, and zone.
 <3> Specify the role node label to add.
-<4> Specify an ARM64 supported Red Hat Enterprise Linux CoreOS (RHCOS) Amazon Machine Image (AMI) for your AWS zone for your OpenShift Container Platform nodes.
+<4> Specify a 64-bit ARM supported Red Hat Enterprise Linux CoreOS (RHCOS) Amazon Machine Image (AMI) for your AWS zone for your OpenShift Container Platform nodes.
```
Updated as follows:

"Specify a Red Hat Enterprise Linux CoreOS (RHCOS) Amazon Machine Image (AMI) for your AWS zone for your {product-title} nodes. The RHCOS AMI must be compatible with the machine architecture."

sr1kar99 commented:
/retest

sr1kar99 commented:
/retest

aleskandro commented on modules/multi-architecture-modify-machine-set-aws.adoc:
```diff
@@ -4,26 +4,23 @@
-= Adding an ARM64 compute machine set to your cluster
-
-To configure a cluster with multi-architecture compute machines, you must create a AWS ARM64 compute machine set. This adds ARM64 compute nodes to your cluster so that your cluster has multi-architecture compute machines.
+After creating a multi-architecture cluster, you can add nodes with different architectures. For example, you can add 64-bit ARM compute machines to a cluster that uses 64-bit ARM control plane machines and already includes 64-bit x86 compute machines. In this case, 64-bit ARM is considered the secondary architecture.
```
```suggestion
After creating a multi-architecture cluster, you can add nodes with different architectures. For example, you can add 64-bit ARM compute machines to a cluster that uses 64-bit AMD control plane machines and already includes 64-bit x86 compute machines. In this case, 64-bit ARM is considered the secondary architecture.
```

sr1kar99 replied:
Updated as per the suggestion.

aleskandro commented on modules/multi-architecture-creating-arm64-bootimage.adoc:
```diff
@@ -5,9 +5,9 @@
 [id="multi-architecture-creating-arm64-bootimage_{context}"]
 
-= Creating an ARM64 boot image using the Azure image gallery
+= Creating a 64-bit ARM boot image using the Azure image gallery
```
I think this also applies for the case of creating an amd64 boot image when the cluster was installed with arm64 CP.

aleskandro replied:
This is resolved by duplicating the bootimage creation page and adding a rocedure for x86

aleskandro commented on modules/multi-architecture-modify-machine-set-aws.adoc:
```diff
@@ -106,28 +108,37 @@ $ oc get configmap/coreos-bootimages \
-<5> Specify an ARM64 supported machine type. For more information, refer to "Tested instance types for AWS 64-bit ARM"
-<6> Specify the zone, for example `us-east-1a`. Ensure that the zone you select offers 64-bit ARM machines.
-<7> Specify the region, for example, `us-east-1`. Ensure that the zone you select offers 64-bit ARM machines.
+<5> Specify a 64-bit ARM supported machine type. For more information, see "Tested instance types for AWS 64-bit ARM"
```
```suggestion
<5> Specify a machine type (with the same architecture as the AMI). For more information, see "Tested instance types for AWS 64-bit ARM"
```

Not sure how to describe this.

sr1kar99 replied:
Updated as per the suggestion.

aleskandro commented on modules/multi-architecture-modify-machine-set-aws.adoc:
```diff
@@ -106,28 +108,37 @@ $ oc get configmap/coreos-bootimages \
 ----
-You can then see your created ARM64 machine set.
++
+The output must include the AWS 64-bit ARM machine set that you created.
```
```suggestion
The output must include the machine set that you created.
```

sr1kar99 replied:
Updated as per the suggestion.

aleskandro commented on modules/multi-architecture-modify-machine-set-gcp.adoc:
```diff
@@ -121,27 +124,35 @@ Use the `project` and `name` parameters from the output to create the path to im
 ----
 <4> Optional: Specify custom metadata in the form of a `key:value` pair. For example use cases, see the GCP documentation for link:https://cloud.google.com/compute/docs/metadata/setting-custom-metadata[setting custom metadata].
-<5> Specify an ARM64 supported machine type. For more information, refer to _Tested instance types for GCP on 64-bit ARM infrastructures_ in "Additional resources".
+<5> Specify a 64-bit ARM supported machine type. For more information, see "Tested instance types for GCP on 64-bit ARM infrastructures".
```
Same issue, can we generalize and discuss about a secondary architecture AND higlighting that instance/machine type and RHCOS image should be for the same arch?

aleskandro commented on modules/multi-architecture-modify-machine-set-gcp.adoc:
```diff
@@ -121,27 +124,35 @@ Use the `project` and `name` parameters from the output to create the path to im
 $ oc get machineset -n openshift-machine-api
 ----
-You can then see your created ARM64 machine set.
+The output must include the GCP 64-bit ARM machine set that you created.
```
```suggestion
The output must include the machine set that you created.
```

sr1kar99 replied:
Updated as per the suggestion.

aleskandro reviewed (commented):
Hi @sr1kar99 i think we are almost ok. I'm just wonder if we can be a bit more generic in the various sections and start talking more about secondary architectures than procedures for having arm64 machines in amd64 CP-based clusters.

Examples are good to cite ARM64 as target secondary architecture, but wherever we can remove the 'example' concept, I would do it.

aleskandro commented:
@jeffdyoung what do you think?

jeffdyoung commented on post_installation_configuration/configuring-multi-arch-compute-machines/creating-multi-arch-compute-nodes-azure.adoc:
```diff
@@ -6,9 +6,13 @@ include::_attributes/common-attributes.adoc[]
+
+After creating a multi-architecture cluster, you can add nodes with different architectures to the cluster. 
+
+For a cluster with 64-bit x86 compute machines, the following procedures explain how to generate a 64-bit ARM boot image and create an Azure compute machine set that uses the 64-bit ARM boot image. You can add compute nodes to your cluster and deploy the 64-bit ARM or 64-bit x86 virtual machines (VM).
```
@sr1kar99 I think we can remove this entire line.

sr1kar99 replied:
Updated as per the suggestion.

sr1kar99 commented:
> > @sr1kar99 only a few small things, otherwise LGTM. The only times we use `arm64/aarch64/amd64/x86_64` is when a command's syntax requires it, or if a cloud provider uses a specific term. Otherwise use `64-bit ARM` and `64-bit x86`.
> 
> Hi @jeffdyoung Thanks for the review. Updated as per the suggestions.
> 
> In the "[Adding a multi-architecture compute machine set to your Azure cluster](https://76246--ocpdocs-pr.netlify.app/openshift-enterprise/latest/post_installation_configuration/configuring-multi-arch-compute-machines/creating-multi-arch-compute-nodes-azure#multi-architecture-modify-machine-set_creating-multi-arch-compute-nodes-azure)" module, we have the following line:
> 
> "To configure an Azure cluster with multi-architecture compute machines, you must create an Azure 64-bit ARM or 64-bit x86 compute machine set that uses the 64-bit ARM or 64-bit x86 boot image."
> 
> Though we have added support for `64-bit x86` arch, we don't have a procedure on "Creating a 64-bit x86 boot image using the Azure image gallery" similar to the existing "[Creating a 64-bit ARM boot image using the Azure image gallery](https://76246--ocpdocs-pr.netlify.app/openshift-enterprise/latest/post_installation_configuration/configuring-multi-arch-compute-machines/creating-multi-arch-compute-nodes-azure#multi-architecture-creating-arm64-bootimage_creating-multi-arch-compute-nodes-azure)" proc.
> 
> Should we modify the existing "Creating a 64-bit ARM boot image using the Azure image gallery" proc to include the support for `64-bit x86` arch or add a separate module?

Hi @jeffdyoung @aleskandro 

I have updated the content based on your suggestions. I too had similar doubts.

I have tried updating the content to sound more general and not specific to any architecture. 

However, the existing "Creating a 64-bit ARM boot image using the Azure image gallery" proc was a bit challenging to update. It contains quite a few commands/examples specific to ARM.

Could you please suggest the generic updates to this procedure?

Also, could you PTAL at the updated content and see if I've missed anything.
Thanks!

sr1kar99 commented on modules/multi-architecture-modify-machine-set.adoc:
```diff
@@ -4,23 +4,29 @@
 .Prerequisites
 
 * You installed the OpenShift CLI (`oc`).
+* You used the installation program to create a 64-bit ARM or 64-bit x86 single-architecture AWS cluster with the multi-architecture installer binary.
```
Updated as per the suggestion.

aleskandro commented on modules/multi-architecture-creating-arm64-bootimage.adoc:
```diff
@@ -5,9 +5,9 @@
 [id="multi-architecture-creating-arm64-bootimage_{context}"]
 
-= Creating an ARM64 boot image using the Azure image gallery
+= Creating a 64-bit ARM or 64-bit AMD boot image using the Azure image gallery
```
```suggestion
= Creating a 64-bit ARM or 64-bit x86 boot image using the Azure image gallery
```

I see the other references for amd64 are 64-bit x86.

sr1kar99 replied:
Yes! Updated it to x86

aleskandro commented on modules/multi-architecture-modify-machine-set-gcp.adoc:
```diff
@@ -4,27 +4,30 @@
 
 * You installed the {oc-first}.
-* You used the installation program to create an AMD64 single-architecture AWS cluster with the multi-architecture installer binary.
+* You used the installation program to create an AMD64 or AMD64 single-architecture AWS cluster with the multi-architecture installer binary.
```
```suggestion
* You used the installation program to create an 64-bit x86 or 64-bit ARM single-architecture AWS cluster with the multi-architecture installer binary.
```

sr1kar99 replied:
Updated as per the suggestion.

sr1kar99 commented:
@jeffdyoung @aleskandro 

FYI: I added a new module for "Creating a 64-bit x86 boot image using the Azure image gallery".
As these are two different use cases, it might be confusing and challenging for users to absorb the content if they are clubbed together. So I create a separate module for x86 use case.

aleskandro commented on modules/multi-architecture-modify-machine-set-gcp.adoc:
```diff
@@ -123,27 +126,35 @@ Use the `project` and `name` parameters from the output to create the path to im
 ----
 <4> Optional: Specify custom metadata in the form of a `key:value` pair. For example use cases, see the GCP documentation for link:https://cloud.google.com/compute/docs/metadata/setting-custom-metadata[setting custom metadata].
-<5> Specify an ARM64 supported machine type. For more information, refer to _Tested instance types for GCP on 64-bit ARM infrastructures_ in "Additional resources".
+<5> Specify a machine type that aligns with the CPU architecture of the chosen AMI. For more information, see "Tested instance types for GCP on 64-bit ARM infrastructures".
```
AMI is AWS-specific terminology.

```suggestion
<5> Specify a machine type that aligns with the CPU architecture of the chosen OS Image. For more information, see "Tested instance types for GCP on 64-bit ARM infrastructures".
```

ref: https://cloud.google.com/compute/docs/images

sr1kar99 replied:
We have only 1 instance of "AMI" in the GCP doc.
I guess this was directly suggested by @jeffdyoung 
Will let him also take a look.

jeffdyoung replied:
yeah, I missed this. GCP shouldn't have "AMI"

sr1kar99 replied:
Updated as per the suggestion.

aleskandro reviewed (commented):
These changes look good to me. Only one nit that can be solved throughout the perr-review process too.

/label qe-approved

aleskandro commented on post_installation_configuration/configuring-multi-arch-compute-machines/creating-multi-arch-compute-nodes-azure.adoc:
```diff
@@ -6,14 +6,18 @@ include::_attributes/common-attributes.adoc[]
 toc::[]
 
-To deploy an Azure cluster with multi-architecture compute machines, you must first create a single-architecture Azure installer-provisioned cluster that uses the multi-architecture installer binary. For more information on Azure installations, see xref:../../installing/installing_azure/installing-azure-customizations.adoc#installing-azure-customizations[Installing a cluster on Azure with customi ...
+To deploy an Azure cluster with multi-architecture compute machines, you must first create a single-architecture Azure installer-provisioned cluster that uses the multi-architecture installer binary. For more information on Azure installations, see xref:../../installing/installing_azure/installing-azure-customizations.adoc#installing-azure-customizations[Installing a cluster on Azure with customi ...
```
```suggestion
To deploy an Azure cluster with multi-architecture compute nodes, you must first create a single-architecture Azure installer-provisioned cluster that uses the multi-architecture installer binary. For more information on Azure installations, see xref:../../installing/installing_azure/installing-azure-customizations.adoc#installing-azure-customizations[Installing a cluster on Azure with customizations]. 
```

We use to call this scenario `Multi-architcture compute nodes clusters`. Leaving this decision up to you

sr1kar99 replied:
This is a really good point. I also notice that we have used "machines" interchangeably with "nodes" in many places. I'd like to take a closer look at this and work on maintaining a consistent terminology across the docs.
For now, I'd leave this as is. But I pick this up on priority in the future.

sr1kar99 commented:
/label peer-review-needed

jeffdyoung commented on modules/multi-architecture-creating-x86_64-bootimage.adoc:
@sr1kar99 sorry, I missed this as well. `--architecture x86_64` should be `--architecture x64` 

Everything else in the command is correct:
```
$ az sig image-definition create --resource-group ${RESOURCE_GROUP} --gallery-name ${GALLERY_NAME} --gallery-image-definition rhcos-x86_64 --publisher RedHat --offer x86_64 --sku x86_64 --os-type linux --architecture x64 --hyper-v-generation V2
```

sr1kar99 replied:
Updated as per the suggestion.

mburke5678 commented on modules/multi-architecture-creating-x86_64-bootimage.adoc:
```diff
@@ -0,0 +1,133 @@
+----
+$ az login
+----
+. Create a storage account and upload the `x86_64` virtual hard disk (VHD) to your storage account. The {product-title} installation program creates a resource group, however, the boot image can also be uploaded to a custom named resource group:
```
@sr1kar99 I assume you saw this suggestion?

mburke5678 commented on modules/multi-architecture-creating-x86_64-bootimage.adoc:
I like consistency. 
```suggestion
. Generate a shared access signature (SAS) token. Use this token to upload the {op-system} VHD to your storage container by running the following commands:
```

sr1kar99 replied:
Updated as per the suggestion.

mburke5678 commented on modules/multi-architecture-creating-x86_64-bootimage.adoc:
```suggestion
. Copy the {op-system} VHD into the storage container by running the following commands:
```

sr1kar99 replied:
Updated as per the suggestion.

mburke5678 commented on modules/multi-architecture-creating-x86_64-bootimage.adoc:
```suggestion
You can check the status of the copying process by running the following commands:```

sr1kar99 replied:
Updated as per the suggestion.

mburke5678 commented on modules/multi-architecture-creating-x86_64-bootimage.adoc:
```suggestion
<1> If the `status` parameter displays `success`, the copying process is complete.
```

sr1kar99 replied:
Updated as per the suggestion.

sr1kar99 commented on modules/multi-architecture-creating-x86_64-bootimage.adoc:
```diff
@@ -0,0 +1,133 @@
+----
+$ az login
+----
+. Create a storage account and upload the `x86_64` virtual hard disk (VHD) to your storage account. The {product-title} installation program creates a resource group, however, the boot image can also be uploaded to a custom named resource group:
```
Updated as per the suggestion.

mburke5678 commented on modules/multi-architecture-creating-x86_64-bootimage.adoc:
> `rhcos-x86_64` is the name of the image definition.
Should this be a call out <1> in the command?

If so, 
```suggestion
Use the image gallery to create an image definition by running the following command:
```

sr1kar99 replied:
Updated as per the suggestion.

mburke5678 commented on modules/multi-architecture-modify-machine-set-aws.adoc:
```diff
@@ -4,28 +4,30 @@
+After creating a multi-architecture cluster, you can add nodes with different architectures.
 
-To configure a cluster with multi-architecture compute machines, you must create a AWS ARM64 compute machine set. This adds ARM64 compute nodes to your cluster so that your cluster has multi-architecture compute machines.
+Following are the examples of adding multi-architecture compute machines to a multi-architecture cluster:
```
```suggestion
You can add the following multi-architecture compute machines to a multi-architecture cluster:

*  Adding 64-bit x86 compute machines to a cluster that uses 64-bit ARM control plane machines and already includes 64-bit ARM compute machines. In this case, 64-bit x86 is considered the secondary architecture.
* 64-bit ARM compute machines. You can add 64-bit ARM compute machines  to a cluster that uses 64-bit x86 control plane machines and already includes 64-bit x86 compute machines. In this case, 64-bit ARM is considered the secondary architecture.
* 64-bit ARM compute machines. You can add 64-bit ARM compute machines to a cluster that uses 64-bit x86 control plane machines and already includes 64-bit x86 compute machines. In this case, 64-bit ARM is considered the secondary architecture.
```

sr1kar99 replied:
Updated as follows:

"You can add multi-architecture compute machines to a multi-architecture cluster in the following ways:"

mburke5678 commented on modules/multi-architecture-modify-machine-set-gcp.adoc:
```diff
@@ -4,27 +4,30 @@
-To configure a cluster with multi-architecture compute machines, you must create a GCP ARM64 compute machine set. This adds ARM64 compute nodes to your cluster.
+After creating a multi-architecture cluster, you can add nodes with different architectures. 
+
+Following are the examples of adding multi-architecture compute machines to a multi-architecture cluster:
```
If you [changed the same text previously](https://github.com/openshift/openshift-docs/pull/76246/files#r1647832286), change it here also. Is this worth a snippet?

sr1kar99 replied:
You can add multi-architecture compute machines to a multi-architecture cluster in the following ways:

mburke5678 commented on modules/multi-architecture-modify-machine-set-gcp.adoc:
```diff
@@ -4,27 +4,30 @@
 +
 --
-.Sample GCP YAML compute machine set to deploy an ARM64 compute node
+.Example `MachineSet` object for a GCP 64-bit ARM or 64-bit x86 compute node:
```
```suggestion
.Example `MachineSet` object for a GCP 64-bit ARM or 64-bit x86 compute node
```

sr1kar99 replied:
Updated as per the suggestion.

mburke5678 commented on modules/multi-architecture-modify-machine-set-gcp.adoc:
```diff
@@ -123,27 +126,35 @@ Use the `project` and `name` parameters from the output to create the path to im
 ----
 <4> Optional: Specify custom metadata in the form of a `key:value` pair. For example use cases, see the GCP documentation for link:https://cloud.google.com/compute/docs/metadata/setting-custom-metadata[setting custom metadata].
-<5> Specify an ARM64 supported machine type. For more information, refer to _Tested instance types for GCP on 64-bit ARM infrastructures_ in "Additional resources".
+<5> Specify a machine type that aligns with the CPU architecture of the chosen OS image. For more information, see "Tested instance types for GCP on 64-bit ARM infrastructures".
```
I agree with this one. There are only 6 instances of _OS image_ in the docs vs 43 of _operating system image_,

```suggestion
<5> Specify a machine type that aligns with the CPU architecture of the chosen operating system image. For more information, see "Tested instance types for GCP on 64-bit ARM infrastructures".
```

mburke5678 commented on modules/multi-architecture-modify-machine-set.adoc:
```diff
@@ -4,25 +4,32 @@
-$ oc create -f arm64-machine-set-0.yaml
-----
-.Sample YAML compute machine set with `arm64` boot image
+.Example `MachineSet` object for an Azure 64-bit ARM or 64-bit x86 compute node:
```
```suggestion
.Example `MachineSet` object for an Azure 64-bit ARM or 64-bit x86 compute node
```

sr1kar99 replied:
Updated as per the suggestion.

mburke5678 commented on modules/multi-architecture-modify-machine-set.adoc:
```diff
@@ -4,25 +4,32 @@
-----
-.Sample YAML compute machine set with `arm64` boot image
+.Example `MachineSet` object for an Azure 64-bit ARM or 64-bit x86 compute node:
 +
```
```suggestion

```

mburke5678 commented on modules/multi-architecture-modify-machine-set-aws.adoc:
```diff
@@ -4,28 +4,30 @@
-$ oc create -f aws-arm64-machine-set-0.yaml
-----
-.Sample YAML compute machine set to deploy an ARM64 compute node
+.Example `MachineSet` object for an AWS 64-bit ARM or x86 compute node:
```
Don't need the colon here.
```suggestion
.Example `MachineSet` object for an AWS 64-bit ARM or x86 compute node
```

mburke5678 commented:
@sr1kar99 A few comments. Otherwise LGTM.

sr1kar99 commented on modules/multi-architecture-modify-machine-set.adoc:
```diff
@@ -4,25 +4,32 @@
+After creating a multi-architecture cluster, you can add nodes with different architectures. 
 
-To add ARM64 compute nodes to your cluster, you must create an Azure compute machine set that uses the ARM64 boot image. To create your own custom compute machine set on Azure, see "Creating a compute machine set on Azure".
+Following are the examples of adding multi-architecture compute machines to a multi-architecture cluster:
```
You can add multi-architecture compute machines to a multi-architecture cluster in the following ways:

sr1kar99 commented:
/label merge-review-needed

skopacz1 commented on modules/multi-architecture-creating-x86_64-bootimage.adoc:
```diff
@@ -0,0 +1,137 @@
+----
+$ az login
+----
+. Create a storage account and upload the `x86_64` virtual hard disk (VHD) to your storage account. The {product-title} installation program creates a resource group. However, the boot image can also be uploaded to a custom named resource group:
```
I think this step still needs to mention running a command, even if it's not the last sentence in the step:
```suggestion
. Create a storage account and upload the `x86_64` virtual hard disk (VHD) to your storage account by running the following command. The {product-title} installation program creates a resource group. However, the boot image can also be uploaded to a custom named resource group:
```

sr1kar99 replied:
Updated as per the suggestion.

skopacz1 commented on modules/multi-architecture-creating-x86_64-bootimage.adoc:
```diff
@@ -0,0 +1,137 @@
+----
+$ az storage container create -n ${CONTAINER_NAME} --account-name ${STORAGE_ACCOUNT_NAME}
+----
+. You must use the {product-title} installation program JSON file to extract the URL and `x86_64` VHD name:
```
The first sentence of each step needs to be in the imperative mood [per the ISG](https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#writing-steps):
```suggestion
. Use the {product-title} installation program JSON file to extract the URL and `x86_64` VHD name:
```

sr1kar99 replied:
Updated as per the suggestion.

skopacz1 commented on modules/multi-architecture-creating-x86_64-bootimage.adoc:
Just checking - is this meant to be a separate action from step 7? It seems like it's a significant enough action, in which case it should be its own step IMO:
```suggestion
. Use the image gallery to create an image definition by running the following command:
```

sr1kar99 replied:
Makes sense to keep it as a separate step.

skopacz1 commented on modules/multi-architecture-creating-x86_64-bootimage.adoc:
```suggestion
. Use the `RHCOS_VHD_URL` file, your storage account, resource group, and image gallery to create an image version by running the following command:
```

sr1kar99 replied:
Updated as per the suggestion.

skopacz1 commented on modules/multi-architecture-creating-x86_64-bootimage.adoc:
Same comment as above about the first sentence being written in the imperative mood. Here's one way to rewrite this but keep all of the same information:
```suggestion
. Access the ID of the generated `x86_64` boot image by running the following command:
```

If the "you can" is meant to indicate that this is optional, I'd add that at the beginning of the step:
```suggestion
. Optional: Access the ID of the generated `x86_64` boot image by running the following command:
```

sr1kar99 replied:
Updated as per the suggestion.

skopacz1 commented on modules/multi-architecture-modify-machine-set-aws.adoc:
```diff
@@ -108,28 +110,37 @@ $ oc get configmap/coreos-bootimages \
+----
+$ oc create -f <file_name> <1>
+----
+<1> Replace `<file_name>` with the name of the YAML file with compute machine set configuration. For example: `aws-arm64-machine-set-0.yaml`, or `aws-amd64-machine-set-0.yaml`
```
I would add a period at the end of the second sentence:
```suggestion
<1> Replace `<file_name>` with the name of the YAML file with compute machine set configuration. For example: `aws-arm64-machine-set-0.yaml`, or `aws-amd64-machine-set-0.yaml`.
```

sr1kar99 replied:
Updated as per the suggestion.

skopacz1 commented on modules/multi-architecture-modify-machine-set.adoc:
```diff
@@ -4,26 +4,32 @@
 
 * You installed the OpenShift CLI (`oc`).
+* You created a 64-bit ARM or 64-bit x86 boot image.
+* You used the installation program to create a 64-bit ARM or 64-bit x86 single-architecture AWS cluster with the multi-architecture installer binary.
```
just checking - this module seems to focus on Azure, should this prereq mention AWS?

sr1kar99 replied:
This is a great catch! I copied the pre-reqs but missed to update the "AWS" instances.

skopacz1 commented on post_installation_configuration/configuring-multi-arch-compute-machines/creating-multi-arch-compute-nodes-ibm-power.adoc:
```diff
@@ -1,8 +1,8 @@
-include::_attributes/common-attributes.adoc[]
 [id="creating-multi-arch-compute-nodes-ibm-power"]
-= Creating a cluster with multi-architecture compute machines on {ibm-power-title}
+= Creating a cluster with multi-architecture compute machines on IBM Power
```
Is there a reason you're removing the attribute from this heading? If there's no specific reason, the attribute should be used.

sr1kar99 replied:
I noticed that these attributes were not resolved properly in the preview. The issue persisted even after multiple retries, so I replaced the attributes with their actual values.

skopacz1 commented on post_installation_configuration/configuring-multi-arch-compute-machines/creating-multi-arch-compute-nodes-ibm-z-kvm.adoc:
```diff
@@ -1,8 +1,8 @@
-include::_attributes/common-attributes.adoc[]
 [id="creating-multi-arch-compute-nodes-ibm-z-kvm"]
-= Creating a cluster with multi-architecture compute machines on {ibm-z-title} and {ibm-linuxone-title} with {op-system-base} KVM
+= Creating a cluster with multi-architecture compute machines on IBM Z and IBM LinuxONE with {op-system-base} KVM
```
Is there a reason you're removing the attributes from this heading? If there's no specific reason, the attributes should be used.

sr1kar99 replied:
I noticed that these attributes were not resolved properly in the preview. The issue persisted even after multiple retries, so I replaced the attributes with their actual values.

skopacz1 commented on post_installation_configuration/configuring-multi-arch-compute-machines/creating-multi-arch-compute-nodes-ibm-z.adoc:
```diff
@@ -1,8 +1,8 @@
-include::_attributes/common-attributes.adoc[]
 [id="creating-multi-arch-compute-nodes-ibm-z"]
-= Creating a cluster with multi-architecture compute machines on {ibm-z-title} and {ibm-linuxone-title} with z/VM
+= Creating a cluster with multi-architecture compute machines on IBM Z and IBM LinuxONE with z/VM
```
Is there a reason you're removing the attributes from this heading? If there's no specific reason, the attributes should be used.

sr1kar99 replied:
I noticed that these attributes were not resolved properly in the preview. The issue persisted even after multiple retries, so I replaced the attributes with their actual values.

skopacz1 reviewed (commented):
Left a few comments, otherwise looks good to me! Let me know when they're addressed and I can merge.

skopacz1 commented:
/cherrypick enterprise-4.16
